bin/improver-probabilities-to-realizations

```diff
@@ -45,26 +47,100 @@ def main():
+                       help='The option used to create ensemble realizations '
+                       'from percentiles by rebadging the input '
+                       'percentiles.')
 
+    # If reordering, we need a raw ensemble forecast.
+    reordering = parser.add_argument_group(
+        'Reordering options', 'Options for reordering the input percentiles '
+        'using the raw ensemble forecast as required to create ensemble '
+        'realizations.')
+    reordering.add_argument('--raw_forecast_filepath',
+                            metavar='RAW_FORECAST_FILE',
+                            help='A path to an raw forecast NetCDF file to be '
+                            'processed. This option is compulsory, if the '
+                            'reordering option is selected.')
+    reordering.add_argument(
+        '--random_seed', default=None,
+        help='Option to specify a value for the random seed for testing '
+             'purposes, otherwise, the default random seed behaviour is '
+             'utilised. The random seed is used in the generation of the '
+             'random numbers used for splitting tied values '
+             'within the raw ensemble, so that the values from the input '
+             'percentiles can be ordered to match the raw ensemble.')
+
+    args = parser.parse_args()
+
+    # CLI argument checking:
+    # Can only do one of reordering or rebadging: if options are passed which
+    # correspond to the opposite method, raise an exception.
+    # Note: Shouldn't need to check that both/none are set, since they are
+    # defined as mandatory, but mutually exclusive, options.
+    if args.rebadging:
+        if ((args.raw_forecast_filepath is not None)
+                or (args.random_seed is not None)):
+            parser.wrong_args_error(
+                'raw_forecast_filepath, random_seed', 'rebadging')
+
+    # Process the data
     cube = load_cube(args.input_filepath)
 
-    cube = GeneratePercentilesFromProbabilities().process(
-            cube, no_of_percentiles=args.no_of_realizations)
-    cube = RebadgePercentilesAsRealizations().process(cube)
+    if args.reordering:
+        if args.raw_forecast_filepath is None:
+            message = ("You must supply a raw forecast filepath if using the "
+                       "reording option.")
+            raise ValueError(message)
+        else:
+            raw_forecast = load_cube(args.raw_forecast_filepath)
+        # If no_of_realizations is not given, take the number from the raw
+        # ensemble cube.
+        if args.no_of_realizations is None:
+            try:
+                no_of_realizations = len(
+                    raw_forecast.coord("realization").points)
+            except CoordinateNotFoundError:
+                message = ("The netCDF file from the raw_forecast_filepath "
+                           "must have a realization coordinate.")
+                raise ValueError(message)
```

Comment:
Is there any situation in which the raw ensemble input file does NOT require a realization coordinate?  If so, should this error be triggered earlier / independently of the argument check?

Also, I don't think there's any situation with the reordering option where the number of realizations should be configurable from the command line.  If reordering, we should always be using the number of realizations in the input file.  In which case I think `args.no_of_realizations` should be forbidden when reordering is set.

Reply:
I think we always need a realization coordinate (I think the plugin probably depends on it). I'm not quite sure I understand where you want to move this check to.

For the max-min temperature use case you're right we only want to use the number of raw ensemble members. In general the ECC plugin is more general than that and can reorder more or less percentiles than the number of raw ensemble members. @gavinevans said that Jonathan did request that functionality when he first put the ECC code into IMPROVER, so I'm reluctant to change this interface.